### DISCUSSION TITLE
extract catalog from dagster

### DISCUSSION THREAD

**Discussion by dridk** [dagster-io/dagster#17806] [Ideas]
In all our Dagster assets, we use Table Schema to document our assets. 
I'd like to to extract this information to build a data catalog (a simple parquet file). 
Do you know where I can get this information ? I guess it is in a sqlite database somewhere. 

![image](https://github.com/dagster-io/dagster/assets/1911063/203ab777-2e75-44c6-b6c8-329619416f23)

A good feature would be to have an endpoint API to display all the metadata and use it as a catalog. 

Exemples: 

```

curl http://server_dagster/schema/ 

[
{
    "assets": "stocks",
     "columns": [
      {"name", "string", "description" }
    ]

}
]


```

**Comment by jamiedemaria** [comment]
hey @dridk  how are you setting the Table Schema so that it shows up in the UI? That will affect how you can access it later

**Reply by dridk** [reply]
Yes I m using dagster table shema

**Reply by jamiedemaria** [reply]
ok - but are you returning it as metadata? something else?

**Reply by dridk** [reply]
Sorry, I misread your answer. 
I m using it as metadata like this. Is there another way? 

```python
@asset
def test(): 
      df = pd.DataFrame(..)

     return Output(df, metadata={ "schema": TableShema .... })

```

**Reply by jamiedemaria** [reply]
No problem. I think you should be able to use the code in [this discussion ](https://github.com/dagster-io/dagster/discussions/14338) to get the metadata for all of the assets in your project
